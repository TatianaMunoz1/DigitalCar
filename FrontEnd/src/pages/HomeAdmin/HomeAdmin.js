import React, { useState } from "react";
import styles from "./HomeAdmin.module.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronDown, faChevronUp } from "@fortawesome/free-solid-svg-icons";
import Button from "../../components/Button/Button";
import ModifyProductPolicySection from "../../components/ModifyProductPolicySection/ModifyProductPolicySection";
import Administration from "../../components/Administration/Administration";
import Subheader from "../../components/Subheader/Subheader";

export default function HomeAdmin(props) {
  const [productManagementSelected, setProductManagementSelected] =
    useState(true);
  const [productManagementMobileSelected, setProductManagementMobileSelected] =
    useState(false);
  const [productPolicyMobileSelected, setProductPolicyMobileSelected] =
    useState(false);
  const [productPolicySelected, setProductPolicySelected] = useState(false);

  return (
    <div className={styles.container}>
      <div className={styles.container_header}>
        <Subheader
          title="Administración"
          subtitle={
            productPolicySelected || productPolicyMobileSelected
              ? "POLÍTICAS DEL PRODUCTO"
              : "CARGAR NUEVO VEHÍCULO"
          }
          link={"/"}
        />
      </div>

      <div className={styles.container_forms_mobile}>
        <div>
          <button
            type="button"
            className={
              productManagementMobileSelected
                ? styles.button_1_mobile_selected
                : styles.button_1_mobile
            }
            onClick={() => {
              setProductManagementMobileSelected(
                !productManagementMobileSelected
              );
              setProductPolicyMobileSelected(false);
            }}
          >
            <span className={styles.btn_content}>
              CARGAR NUEVO VEHÍCULO
              <FontAwesomeIcon
                icon={
                  productManagementMobileSelected ? faChevronUp : faChevronDown
                }
              />
            </span>
          </button>
          {productManagementMobileSelected && <Administration user={props.user}/>}
        </div>
        <div>
          <button
            type="button"
            className={
              productPolicyMobileSelected
                ? styles.button_2_mobile_selected
                : styles.button_2_mobile
            }
            onClick={() => {
              setProductPolicyMobileSelected(!productPolicyMobileSelected);
              setProductManagementMobileSelected(false);
            }}
          >
            <span className={styles.btn_content}>
              POLÍTICAS DEL PRODUCTO
              <FontAwesomeIcon
                icon={productPolicyMobileSelected ? faChevronUp : faChevronDown}
              />
            </span>
          </button>
          {productPolicyMobileSelected && <ModifyProductPolicySection user={props.user}/>}
        </div>
      </div>

      <div className={styles.container_forms_tablet_desktop}>
        <button
          type="button"
          className={
            productManagementSelected
              ? styles.button_1_selected
              : styles.button_1
          }
          onClick={() => {
            setProductManagementSelected(true);
            setProductPolicySelected(false);
          }}
        >
          CARGAR NUEVO VEHÍCULO
        </button>
        <button
          type="button"
          className={
            productPolicySelected ? styles.button_2_selected : styles.button_2
          }
          onClick={() => {
            setProductPolicySelected(true);
            setProductManagementSelected(false);
          }}
        >
          POLÍTICAS DEL PRODUCTO
        </button>
        {productManagementSelected && <Administration user={props.user}/>}
        {productPolicySelected && <ModifyProductPolicySection user={props.user}/>}
      </div>
    </div>
  );
}
