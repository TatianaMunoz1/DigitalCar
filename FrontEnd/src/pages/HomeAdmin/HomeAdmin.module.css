.container {
  background-color: #000000;
  padding: 250px 15px 130px 15px;
  min-height: 100vh;
  height: 100%;
}

.container_header {
  background: #000;
  position: absolute;
  top: 0px;
  left: 0px;
  height: 90px;
  width: 100%;
}
.button_1_mobile,
.button_2_mobile {
  color: #000000;
  font-size: 16px;
  padding: 20px 10px;
  width: 100%;
  background: #dbdbdb;
  border: none;
  cursor: pointer;
  display: inline;
}
.container_forms {
  position: relative;
}
.button_1_mobile:hover,
.button_2_mobile:hover {
  color: #ffa25a;
  transition: all 0.3s ease-out;
}
.button_1_mobile_selected,
.button_2_mobile_selected {
  color: #ff7e19;
  font-size: 16px;
  width: 100%;
  font-weight: 600;
  padding: 20px 10px;
  background: #f0f0f0;
  border: none;
  cursor: pointer;
}
.container_forms_tablet_desktop {
  display: none;
}
.container_forms_mobile {
  display: flex;
  flex-direction: column;
  gap: 15px;
}
.btn_content {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 10px;
  text-align: center;
}

/*-----TABLET-----*/
@media (min-width: 760px) and (max-width: 1280px) {
  .container {
    padding: 250px 50px 130px 50px;
  }
  .button_1,
  .button_2 {
    color: #000000;
    font-size: 16px;
    padding: 20px 8px;
    width: 230px;
    background: #dbdbdb;
    border: none;
    border-radius: 8px 8px 0px 0px;
    cursor: pointer;
  }
  .container_forms_tablet_desktop {
    display: contents;
  }
  .container_forms_mobile {
    display: none;
  }
  .button_1:hover,
  .button_2:hover {
    color: #000000;
    background: #c5c5c5;
    transition: all 0.3s ease-out;
  }
  .button_1_selected,
  .button_2_selected {
    color: #000000;
    font-size: 16px;
    font-weight: 600;
    padding: 20px 10px;
    width: 230px;
    background: #f0f0f0;
    border: none;
    border-radius: 8px 8px 0px 0px;
  }
}

/*-----DESKTOP-----*/
@media (min-width: 1281px) {
  .container {
    padding: 250px 80px 130px 80px;
  }
  .button_1,
  .button_2 {
    color: #000000;
    font-size: 16px;
    padding: 22px 8px;
    width: 230px;
    background: #dbdbdb;
    border: none;
    border-radius: 8px 8px 0px 0px;
    cursor: pointer;
  }
  .container_forms_tablet_desktop {
    display: contents;
  }
  .container_forms_mobile {
    display: none;
  }
  .button_1:hover,
  .button_2:hover {
    color: #000000;
    background: #c5c5c5;
    transition: all 0.3s ease-out;
  }
  .button_1_selected,
  .button_2_selected {
    color: #000000;
    font-size: 16px;
    font-weight: 600;
    padding: 22px 8px;
    width: 230px;
    background: #f0f0f0;
    border: none;
    border-radius: 8px 8px 0px 0px;
  }
}
