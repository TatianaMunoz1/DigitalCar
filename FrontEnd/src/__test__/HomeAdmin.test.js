import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import { shallow, mount, mock } from "enzyme";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import HomeAdmin from '../pages/HomeAdmin/HomeAdmin'

describe("Button", () => {
    const data = {
        title:"Gestión de vehículos",
        link:"/administration",
        btnStyle:"btn--void",
        btnSize:"btn--medium"
    };
    const history = createMemoryHistory();
    

    test("render home admin ", () => {
        const wrapper = shallow(<HomeAdmin datos={data}/>)
        expect(wrapper).toMatchSnapshot();
    });

    test("must display button 1", () => {
      render(
        <Router history={history}>
          <HomeAdmin {...data} />
        </Router>
      );
      const element = screen.findByText("button")
      expect(element).toBeInTheDocument();
      expect(element).toHaveClass("btn");
      expect(element).toHaveClass(data.btnStyle);
      expect(element).toHaveClass(data.btnSize);
    });
});