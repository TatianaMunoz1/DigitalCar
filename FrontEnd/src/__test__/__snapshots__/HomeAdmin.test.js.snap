// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render home admin  1`] = `ShallowWrapper {}`;
